libtransmission/torrent.c

```diff
@@ -2328,6 +2328,8 @@ void tr_torrentRecheckCompleteness(tr_torrent* tor)
 
             if (tr_sessionIsTorrentDoneScriptEnabled(tor->session))
             {
+                tr_torrentSave(tor);
```

Comment:
Since we're saving the torrent here, it doesn't look like a good idea mark it as dirty 6 lines down.

Reply:
Hmm good catch. How about 5082393d925f52a097112ced01fdc9506aecfc9d